tls/s2n_client_hello.c

```diff
@@ -603,3 +603,54 @@ int s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, u
 
     return S2N_SUCCESS;
 }
+
+static int s2n_client_hello_get_raw_extension(uint16_t expected_extension_type, struct s2n_blob *raw_extensions, struct s2n_blob *extension)
+{
+    POSIX_ENSURE_REF(raw_extensions);
+    POSIX_ENSURE_REF(extension);
+
+    struct s2n_stuffer raw_extensions_stuffer = { 0 };
+    POSIX_GUARD(s2n_stuffer_init(&raw_extensions_stuffer, raw_extensions));
+    POSIX_GUARD(s2n_stuffer_skip_write(&raw_extensions_stuffer, raw_extensions->size));
+
+    while(s2n_stuffer_data_available(&raw_extensions_stuffer) > 0) {
+        uint16_t extension_type;
```

Comment:
Always zero-init your variables. Some very old compilers don't automatically zero-init, and it's bitten us before.
```suggestion
    while(s2n_stuffer_data_available(&raw_extensions_stuffer) > 0) {
        uint16_t extension_type = 0;
```

Reply:
Sounds good